### DISCUSSION TITLE
Exception: no disk founded

### DISCUSSION THREAD

**Discussion by Toktatop** [CosmosOS/Cosmos#2730] [Q&A] [answered]
i have geted answer: (You cant use the VFS on real hardware since its a virtual filesystem)  but what can be used besides VFS

why when I run my OS on a real computer via flash drive I get an error: (Exception: disk not found)
here my code:
`using System;
using System.Collections.Generic;
using System.Linq.Expressions;
using System.Text;
using Sys = Cosmos.System;

namespace test.os
{
    public class Kernel : Sys.Kernel
    {
        Sys.FileSystem.CosmosVFS fs;
        string crd = @"0:\";
        protected override void BeforeRun()
        {
            fs = new Sys.FileSystem.CosmosVFS();
            Sys.FileSystem.VFS.VFSManager.RegisterVFS(fs);
            clear();
            Console.WriteLine("type 'help' for help");
        }

        protected override void Run()
        {
            Console.Write(">> ");
            var input = Console.ReadLine();

            // charge place
            if (input == "shutdown")
            {
                Cosmos.System.Power.Shutdown();
            }

            else if (input == "reboot")
            {
                Cosmos.System.Power.Reboot();
            }

            // time place
            else if (input == "datetime")
            {
                Console.WriteLine(DateTime.Now);
            }

            // help place
            else if (input == "help") { 
                Console.ForegroundColor = ConsoleColor.Green;
                Console.WriteLine("datetime\nshutdown\nreboot\ncls\ncalc\ndir");
                Console.ForegroundColor= ConsoleColor.White;
            }

            // file system place
            else if (input == "dir")
            {
                try
                {
                    var directory_list = Sys.FileSystem.VFS.VFSManager.GetDirectoryListing(crd);
                    foreach (var directoryEntry in directory_list)
                    {
                        try
                        {
                            var entry_type = directoryEntry.mEntryType;
                            if (entry_type == Sys.FileSystem.Listing.DirectoryEntryTypeEnum.File)
                            {
                                Console.ForegroundColor = ConsoleColor.Magenta;
                                Console.WriteLine("| <File>       " + directoryEntry.mName);
                                Console.ForegroundColor = ConsoleColor.White;
                            }
                            if (entry_type == Sys.FileSystem.Listing.DirectoryEntryTypeEnum.Directory)
                            {
                                Console.ForegroundColor = ConsoleColor.Blue;
                                Console.WriteLine("| <Directory>      " + directoryEntry.mName);
                                Console.ForegroundColor = ConsoleColor.White;
                            }
                        }
                        catch (Exception e)
                        {
                            Console.WriteLine("Error: Directory not found");
                            Console.WriteLine(e.ToString());
                        }

                    }
                }
                catch (Exception ex)
                {
                    Console.WriteLine(ex.ToString());
                }
            }

            // text place
            else if (input == "cls")
            {
                clear();
            }

            // proggrams
            // calculator
            else if (input == "calc")
            {
                try
                {
                    Console.Write("Enter the first number>> ");
                    decimal num1 = Convert.ToDecimal(Console.ReadLine());

                    Console.Write("Enter the second number>> ");
                    decimal num2 = Convert.ToDecimal(Console.ReadLine());

                    Console.Write("Enter the operator (+, -, *, /)>> ");
                    char op = Convert.ToChar(Console.ReadLine());

                    decimal result;

                    switch (op)
                    {
                        case '+':
                            result = num1 + num2;
                            break;
                        case '-':
                            result = num1 - num2;
                            break;
                        case '*':
                            result = num1 * num2;
                            break;
                        case '/':
                            if (num2 == 0)
                            {
                                throw new DivideByZeroException("Cannot divide by zero.");
                            }
                            result = num1 / num2;
                            break;
                        default:
                            throw new ArgumentException("Invalid operator.");
                    }

                    Console.WriteLine($"Result: {result}");
                }
                catch (FormatException)
                {
                    Console.ForegroundColor = ConsoleColor.Red;
                    Console.WriteLine("Invalid input format. Please enter valid numbers.");
                    Console.ForegroundColor = ConsoleColor.White;
                }
                catch (DivideByZeroException ex)
                {
                    Console.ForegroundColor = ConsoleColor.Red;
                    Console.WriteLine(ex.Message);
                    Console.ForegroundColor = ConsoleColor.White;
                }
                catch (ArgumentException ex)
                {
                    Console.ForegroundColor = ConsoleColor.Red;
                    Console.WriteLine(ex.Message);
                    Console.ForegroundColor = ConsoleColor.White;
                }
                catch (Exception ex)
                {
                    Console.ForegroundColor= ConsoleColor.Red;
                    Console.WriteLine("An error occurred: " + ex.Message);
                    Console.ForegroundColor= ConsoleColor.White;
                }
            }

            // help for user place
            else if (input == "") { }

            else if (input == "beep")
            {
                Console.Beep();
            }

            else
            { 
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("error unknown command: " + input);
                Console.ForegroundColor = ConsoleColor.White;
            }
                    
        }
        public void clear()
        {
            Console.Clear();
            Console.BackgroundColor = ConsoleColor.Blue;
            Console.WriteLine("test.os + time: ( " + DateTime.Now + " )");
            Console.BackgroundColor = ConsoleColor.Black;
        }
    }
}
`

**Comment by 9xbt** [comment]
You cant use the VFS on real hardware since its a **virtual** filesystem

**Reply by 9xbt** [reply]
Well you can but it requires very specific specs which are really rare

**Reply by 9xbt** [reply]
Not really a Cosmos issue but its your hardware that doesnt support the VFS

**Reply by Toktatop** [reply]
what can be used besides VFS

**Reply by retrievaaaa** [reply, marked as answer]
no, this is plain wrong. A VFS is an abstraction layer and does not require some "specific" hardware to be emulated. The only problem is that the Cosmos VFS and the disk drivers are just not that good.

There is no SATA & NVMe Driver, so you are gonna have to connect your drive using IDE. Also, you need to properly format your drive to be recognizable by Cosmos.

**Reply by 9xbt** [reply]
Well if it doesn't support SATA or NVMe then you cant use SATA or NVMe so you need a computer that has IDE, that's what i was trying to say

**Reply by 9xbt** [reply]
I didn't really want to complicate this but whatever
